src/test/throttler/queue.spec.ts

```diff
@@ -0,0 +1,29 @@
+import * as chai from "chai";
+
+const { expect } = chai;
+
+import Queue from "../../throttler/queue";
+
+describe("Queue", () => {
+  it("should have default name of queue", () => {
+    const queue = new Queue({});
+    expect(queue.name).to.equal("queue");
+  });
+
+  it("should be first-in-first-out", () => {
+    const queue = new Queue({});
+    expect(queue.hasWaitingTask()).to.equal(false);
+
+    queue.total++;
```

Comment:
_huge_ -1 on this behavior

Use the queue like you _actually_ would to test it. Don't set internal state like this and write assertions against it.